src/components/ContextData/index.jsx

```diff
@@ -0,0 +1,18 @@
+import React,{ createContext, useContext, useState } from 'react'
+import data from '../../data'
+
+export const ContextData = createContext()
+
+export const DataProvider = ({ initalState = {}, children }) => {
+  const [values, setValues] = useState(initalState)
+
+  return (
+    <ContextData.Provider value={{ ...data, values, setValues }}>
+      {children}
+    </ContextData.Provider>
+  );
+}
+
+export const DataConsumer = ContextData.Consumer;
```

Comment:
I don't tend to use this but I make it available anyway. The `useContext` hook gives a better development experience at providing the same thing.

For comparison: 

with Consumer:

```js
const SomeComponent = () => {
    return  (
        <DataConsumer>
            {({ config, hexGrid, setValues, values }) => {
                return <div>{ values }</div>
            }}
        </DataConsumer>
    )
}
```

with hook:


```js
const SomeComponent = () => {
    const { config, hexGrid, setValues, values } = useData()
    return  (
        <div>{ values }</div>
    )
}
```